atom/browser/api/atom_api_global_shortcut.cc

```diff
@@ -38,6 +39,19 @@ void GlobalShortcut::OnKeyPressed(const ui::Accelerator& accelerator) {
   accelerator_callback_map_[accelerator].Run();
 }
 
+bool GlobalShortcut::RegisterAll(
+    const std::vector<ui::Accelerator>& accelerators,
+    const base::Closure& callback) {
+  for (auto& accelerator : accelerators) {
+    GlobalShortcutListener* listener = GlobalShortcutListener::GetInstance();
+    if (!listener->RegisterAccelerator(accelerator, this))
```

Comment:
If one fails, should we unregister the successful ones?

Otherwise we will be returning `false` as if it failed when in fact some were successfully registered

Reply:
oh yes this is a good point 🧐